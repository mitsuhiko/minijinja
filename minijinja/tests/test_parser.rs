#![cfg(feature = "unstable_machinery")]
use minijinja::machinery::parse;

#[test]
fn test_parser() {
    insta::glob!("parser-inputs/*.txt", |path| {
        let contents = std::fs::read_to_string(path).unwrap();
        let filename = path.file_name().unwrap().to_str().unwrap();
        let ast = parse(&contents, filename, Default::default(), Default::default());
        insta::with_settings!({
            description => contents.trim_end(),
            omit_expression => true,
        }, {
            insta::assert_debug_snapshot!(&ast);
        });
    });
}
