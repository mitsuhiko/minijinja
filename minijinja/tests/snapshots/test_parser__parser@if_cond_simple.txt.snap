---
source: tests/test_parser.rs
expression: "&ast"
input_file: tests/parser-inputs/if_cond_simple.txt

---
Ok(
    Template {
        children: [
            IfCond {
                expr: Var {
                    id: "expr1",
                } @ 1:6-1:11,
                true_body: [
                    EmitRaw {
                        raw: "\n    branch 1\n",
                    } @ 1:14-3:0,
                ],
                false_body: [],
            } @ 1:3-3:11,
            EmitRaw {
                raw: "\n",
            } @ 3:11-4:0,
        ],
    } @ 1:0-4:0,
)
